src/screens/WalletConnect/ConnectedApps/AppListItem.js

```diff
@@ -19,60 +19,175 @@
+
+  const onPressButton = () => {
+    onPress && onPress();
+    setVisibleModal(!visibleModal);
+  };
+
+  const onChangeNetwork = () => {
+    setVisibleNetworkSwitchModal(!visibleNetworkSwitchModal);
+  };
+
+  React.useEffect(() => {
+    DeviceEventEmitter.emit(WALLET_DROPDOWN_REF, Dropdownref);
+  }, [Dropdownref, visibleModal, visibleWalletSwitchModal]);
+
+  React.useEffect(() => {
+    DeviceEventEmitter.emit(WALLET_DROPDOWN_REF, NetworkRef);
+  }, [NetworkRef, visibleNetworkSwitchModal]);
+
+  const onChangeChainSession = (updatedChain: Chain) => {
+    const chainId = mapChainToChainId(updatedChain);
+    if (chainId === connector.chainId) return;
+    updateConnectorSession(connector, { chainId, accounts: connector.accounts });
+  };
+
+  const onChangeSessionAccount = (accountId: string) => {
+    dispatch(switchAccountAction(accountId));
+    updateConnectorSession(connector, { chainId: 1, accounts: [accountId] });
+  };
+
+  const disconnect = () => {
+    const sessionUrl = connector.peerMeta?.url;
+    if (!sessionUrl) {
+      Toast.show({
+        message: t('toast.missingSessionUrl'),
+        emoji: 'hushed',
+        supportLink: true,
+        autoClose: false,
+      });
+      setVisibleWalletSwitchModal(false);
+      return;
+    }
+
+    disconnectSessionByUrl(sessionUrl);
+    setVisibleWalletSwitchModal(false);
+  };
+
   return (
-    <TouchableContainer onPress={onPress}>
-      <IconContainer>{!!iconUrl && <IconImage source={{ uri: iconUrl }} />}</IconContainer>
-
-      <TitleContainer>
-        <Text variant="medium" numberOfLines={1}>{title}</Text>
-        <Text color={colors.secondaryText}>{config?.title}</Text>
-      </TitleContainer>
-
-      <RightAddOn>
-        <Text variant="medium" color={colors.secondaryText}>
-          {t('connected')}
-        </Text>
-      </RightAddOn>
-    </TouchableContainer>
+    <Container>
+      <Line />
+      <TouchableContainer ref={Dropdownref} onPress={onPressButton}>
+        <IconContainer>{!!iconUrl && <IconImage source={{ uri: iconUrl }} />}</IconContainer>
+
+        <TitleContainer>
+          <Text variant="medium" numberOfLines={1}>
+            {title}
+          </Text>
+          <Text color={colors.secondaryText}>{walletData[0]?.label}</Text>
+        </TitleContainer>
+
+        <RightAddOn ref={NetworkRef} onPress={onChangeNetwork}>
+          <Icon name={chain} width={16} />
+          <Text variant="medium" color={colors.basic010} style={{ marginLeft: 5 }}>
+            {config?.title}
+          </Text>
+        </RightAddOn>
+      </TouchableContainer>
+      <ConnectedAppsMenu
+        visible={visibleModal}
+        onHide={setVisibleModal}
+        onSelect={(item: itemProps) => {
+          if (item.value === 'Switch wallet') setVisibleWalletSwitchModal(true);
+          if (item.value === 'Switch network') setVisibleNetworkSwitchModal(true);
+          if (item.value === 'Disconnect') disconnect();
```

Comment:
Would this work for different languages?

Reply:
Yes, this is working in different languages. label showing in app.